GH-749 Add Second Y Axis to Google Charts

dtsnell4 opened the pull request:
Updates Google Bar/Column, Line and Scatter charts to latest Google Material Charts version.  Implements the "axes" and "series" options to create dual-y axes.

Fixes GH-749

Signed-off-by: Dan Snell <Dan.Snell@lexisnexis.com>

dtsnell4 commented on src/google/Bar.js:
```diff
@@ -179,10 +202,26 @@
+//        this.yAxisPrimaryLabel("Year 1 grade");
+//        this.yAxisSecondaryLabel("Year 2 grade");
+//        return this;
+//    };
```
remove comments

GordonSmith replied:
Remove commented code.

dtsnell4 commented:
@GordonSmith @mlzummo @jbrundage Please review.

You can view a working demo with appropriate testData here:  http://rawgit.com/dtsnell4/Visualization/GH-749_testdata/demos/dermatology.html?src/google/Bar

Of note, once Google makes an additional y axis, it wants to make one for every series in the data.  That means if you try to add a second y-axis with our standard ND testData, it make a third y-axis. Technically I believe I could make an axis for any number of series, but the problem is its all or none.  You can't have data with 5 series and only show 2 y-axes.  It will show 5 regardless.  

You can view the charts with our standard data here and see what happens. http://rawgit.com/dtsnell4/Visualization/GH-749_Google_SecondYAxis/demos/dermatology.html?src/google/Bar

Just add yAxisPrimaryLabel and yAxisSecondaryLabel in the property editor.  That extra set of labels on the bottom is for the third data series.

GordonSmith commented:
Lint issues

GordonSmith commented:
@dtsnell4 - I will look at this once I have done the regular chart variant as I want to think about how we specify which data goes to which y axis...

GordonSmith commented:
@dtsnell4 I believe I have fixed the candidate-1.6.6, 1.8.0 and master branches so that they will build consistently on travis, so if you can rebase again it should be good.

dtsnell4 commented:
@GordonSmith I rebased and fixed the lint issues, but still failing.  Additionally, with the new test data rebased, I now need to refactor the Factory data for the Google Bar/Columns to combine them as we did with HPCC's.  Since you are still considering which data goes to which axis, I will wait for you before making those changes.

GordonSmith commented:
They are real errors - you need to add the bar package to:  https://github.com/hpcc-systems/Visualization/blob/master/test/runner_build_nonamd.html#L73
and probably to:  https://github.com/hpcc-systems/Visualization/blob/master/demos/test-no-amd.html#L85

dtsnell4 commented:
@GordonSmith I also added the bar package to dist/hppc-viz-google.js for the Bar, Line and Scatter widgets and unitTesBuildNonAMD passes when I run it locally.

GordonSmith commented:
You will also need to add the line and scatter packages to the non-amd html files...

dtsnell4 commented:
@GordonSmith sorry, its good to go now

GordonSmith commented:
Looks fine - we will probably need to come up with an official way of specifying which data goes to which axis.
